lib/train/platforms/detect/helpers/os_windows.rb

```diff
@@ -4,7 +4,11 @@ module Train::Platforms::Detect::Helpers
   module Windows
     def detect_windows
       res = @backend.run_command('cmd /c ver')
-      return false if res.exit_status != 0 or res.stdout.empty?
+      if res.exit_status != 0 or res.stdout.empty?
```

Comment:
I've added a test for that situation in test/unit/platforms/detect/os_windows_test.rb.

Microsoft OpenSSH tries to run `ver"` with the run_command in line 6 and fails. In that case try to run only `ver` which then works and returns us the version information.

We can optimize that and only have one run_command here if a change in line 6 would accepted

```
res = @backend.run_command('cmd /c ver ')
```

Just add a space after `ver` and then also Microsoft OpenSSH server works with only one run_command.

Reply:
I'm all for adding the trailing space in the detection code.

Reply:
How odd... Good catch. Can you capture this comment and the one on the related test as a comment in the code so we remember this 6+ months from now? Otherwise I expect something like rubocop will try to clean it and we'll pass it through.

Reply:
I think the trailing space version with only one call makes more sense (even tho it doesn't make ANY sense to me).

Reply:
Adding the trailing space is also my preferred way to keep it at only one detection call. I'll update the PR accordingly.

Reply:
Oh I'll try `cmd.exe /c ver` as suggested in https://github.com/PowerShell/Win32-OpenSSH/issues/1373#issuecomment-488802865
It looks like we get some attention in all related repos and find a good solution!